src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -416,6 +425,20 @@ private async Task WaitForClientsDisconnect(HubServiceEndpoint endpoint)
             Log.TimeoutWaitingClientsDisconnect(_logger, endpoint.ToString(), (int)_scaleTimeout.TotalSeconds);
         }
 
+        private IEnumerable<ServiceEndpoint> SingleOrNotSupported(IEnumerable<ServiceEndpoint> endpoints, ServiceMessage message)
+        {
+            var endpointCnt = endpoints.ToList().Count;
+            if (endpointCnt == 1)
+            {
+                return endpoints;
+            }
+            if (endpointCnt == 0)
+            {
+                throw new ArgumentException("Client invocation is not sent because no endpoint is returned from the endpoint router.");
```

Comment:
throw AzureSignalRNoEndpointAvailableException?

Reply:
The exception sounds like endpoints are down but actually results are returned from (custom) router. So suggest current information is clear enough.